### PR TITLE
Remove the cross namespace OwnerReference

### PR DESCRIPTION
**Pull request by ZhiminXiang**
1. Remove the cross-namespace OwnerReference added when copying secret from user's application namespace to Istio gateway namespace
2. Explicitly delete the secret resources when Ingress is being deleted.

https://github.com/knative/serving/issues/7305

### CODE REVIEW

File: pkg/reconciler/ingress/ingress.go [hunk 1]
```diff
@@ -265,6 +262,33 @@ func (r *Reconciler) FinalizeKind(ctx context.Context, ing *v1alpha1.Ingress) pk
...
+			return err
+		}
+		for _, nameNamespace := range nameNamespaces {
+			secrets, err := r.GetSecretLister().Secrets(nameNamespace.Namespace).List(labels.SelectorFromSet(
+				resources.MakeTargetSecretLabels(tls.SecretName, tls.SecretNamespace)))
+			if err != nil {
+				return err
+			}
+			for _, secret := range secrets {
+				if err := r.GetKubeClient().CoreV1().Secrets(secret.Namespace).Delete(secret.Name, &metav1.DeleteOptions{}); err != nil {
```
Comment (vagababov) [inline pkg/reconciler/ingress/ingress.go]: Do we want to continue deleting?

Comment (ZhiminXiang) [reply pkg/reconciler/ingress/ingress.go]: I think we may want. Updated the code to try deleting the rest of secrets when error happens.

**Pull request merged**